cmake/modules/AddSwiftUnittests.cmake

```diff
@@ -43,8 +43,22 @@ function(add_swift_unittest test_dirname)
     set_property(TARGET "${test_dirname}" APPEND_STRING PROPERTY
       LINK_FLAGS " -Xlinker -rpath -Xlinker ${SWIFT_LIBRARY_OUTPUT_INTDIR}/swift/macosx")
   elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-    set_property(TARGET "${test_dirname}" APPEND_STRING PROPERTY
-      LINK_FLAGS " -latomic")
+    if(NOT "${SWIFT_ANDROID_NATIVE_SYSROOT}" STREQUAL "")
```

Comment:
I was trying to keep the surrounding logic the same as much as possible, to avoid raising the SWIFT_HOST_VARIANT/CMAKE_SYSTEM_NAME issue again. Otherwise, I have a third possibility:
```
elseif("${SWIFT_HOST_VARIANT}" STREQUAL "linux" OR "${SWIFT_HOST_VARIANT}" STREQUAL "android")
```
Or alternately, separating these two SWIFT_HOST_VARIANT checks like you do in your example.

Reply:
My idea was keeping the Linux part of the code separate. That way the people dealing with Linux can do their modifications without worrying about that piece of Android code in there. I choose to keep comparing with `CMAKE_SYSTEM_NAME` because all the other checks in this part of the code uses that, so I didn’t want to break the pattern with something like `SWIFT_HOST_VARIANT` by itself, if you prefer to use that instead of checking for `SWIFT_ANDROID_NATIVE_SYSROOT` it is fine by me. The branch that has to be introduced has to be before the Linux branch, sadly, but there's little we can do about that.

Reply:
Done, and uses swift_android_lib_for_arch instead now.